pgmpy/benchmark/CI_benchmark.py

```diff
@@ -0,0 +1,271 @@
+import os
+import warnings
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+
+from pgmpy.benchmark.data_generator_mechanism import DGP_REGISTRY
+from pgmpy.estimators import CITests
+
+warnings.filterwarnings("ignore", message="divide by zero encountered in divide")
+
+DGM_TO_CITESTS = {
+    "linear_gaussian": ["pearsonr", "gcm", "pillai"],
+    "nonlinear_gaussian": ["gcm", "pillai"],
+    "non_gaussian_continuous": ["gcm", "pillai"],
+    "discrete_categorical": [
+        "chi_square",
+        "log_likelihood",
+        "modified_log_likelihood",
+        "pillai",
+    ],
+    "mixed_data": ["pillai"],
+}
+
+dgms = {name: DGP_REGISTRY[name] for name in DGM_TO_CITESTS.keys()}
+
+sample_sizes = [20, 40, 100, 500, 1000]
+n_cond_vars_list = [1, 3]
+noise_levels = [0.5, 1]
+n_repeats = 10
+significance_levels = [0.001, 0.01, 0.1, 1.0]
+ci_tests = {
+    "pearsonr": CITests.pearsonr,
+    "gcm": CITests.gcm,
+    "chi_square": CITests.chi_square,
+    "log_likelihood": CITests.log_likelihood,
+    "modified_log_likelihood": CITests.modified_log_likelihood,
+    "pillai": CITests.pillai_trace,
+}
+
+
+def run_benchmark(
+    dgms,
+    dgm_to_citests,
+    ci_tests,
+    sample_sizes,
+    noise_levels,
+    n_repeats,
+    significance_levels,
+    n_cond_vars_list,
+):
+    results = []
+    for dgm_name, dgm in dgms.items():
+        print(f"\nRunning DGM: {dgm_name}")
+        compatible_tests = dgm_to_citests[dgm_name]
+        for n_cond_vars in n_cond_vars_list:
+            for n in sample_sizes:
+                for noise in noise_levels:
+                    for significance_level in significance_levels:
+                        for effect_type, eff in [("null", 0.0), ("alt", 1.0)]:
+                            for rep in range(n_repeats):
+                                dgm_kwargs = dict(
+                                    n_samples=n,
+                                    effect_size=eff,
+                                    noise_std=noise,
+                                    n_cond_vars=n_cond_vars,
+                                    seed=rep,
+                                    dependent=True if effect_type == "alt" else False,
+                                )
+                                df = dgm(**dgm_kwargs)
+                                z_cols = [
+                                    col
+                                    for col, typ in df.attrs["variable_types"].items()
+                                    if col.startswith("Z")
+                                ]
+                                for test_name in compatible_tests:
+                                    ci_func = ci_tests[test_name]
+                                    accepted = None
+                                    try:
```

Comment:
Why do we need a try-except block here? Shouldn't we already know which tests would work for the given data generating mechanism?